sklearn/mixture/bayesian_mixture.py

```diff
@@ -0,0 +1,689 @@
+# Author: Wei Xue <xuewei4d@gmail.com>
+#         Thierry Guillemot <thierry.guillemot.work@gmail.com>
+# License: BSD 3 clause
+
+import math
+import numpy as np
+from scipy.special import digamma, gammaln
+
+from .base import BaseMixture, _check_shape
+from .gaussian_mixture import _check_precision_matrix
+from .gaussian_mixture import _check_precision_positivity
+from .gaussian_mixture import _compute_log_det_cholesky
+from .gaussian_mixture import _compute_precision_cholesky
+from .gaussian_mixture import _estimate_gaussian_parameters
+from .gaussian_mixture import _estimate_log_gaussian_prob
+from ..utils import check_array
+from ..utils.validation import check_is_fitted
+
+
+def _log_dirichlet_norm(dirichlet_concentration):
+    """Compute the log of the Dirichlet distribution normalization term.
+
+    Parameters
+    ----------
+    dirichlet_concentration : array-like, shape (n_samples,)
+        The parameters values of the Dirichlet distribution.
+
+    Returns
+    -------
+    log_dirichlet_norm : float
+        The log normalization of the Dirichlet distribution.
+    """
+    return (gammaln(np.sum(dirichlet_concentration)) -
+            np.sum(gammaln(dirichlet_concentration)))
+
+
+def _log_wishart_norm(degrees_of_freedom, log_det_precisions_chol, n_features):
+    """Compute the log of the Wishart distribution normalization term.
+
+    Parameters
+    ----------
+    degrees_of_freedom : array-like, shape (n_components,)
+        The parameters values of the Whishart distribution.
+
+    log_det_precision_chol : array-like, shape (n_components,)
+        The determinant of the cholesky decomposition of the precision matrix.
+
+    n_features : int
+        The number of features.
+
+    Return
+    ------
+    log_wishart_norm : array-like, shape (n_components,)
+        The log normalization of the Wishart distribution.
+    """
+    # To simplify the computation we have removed the np.log(np.pi) term
+    return -(degrees_of_freedom * log_det_precisions_chol +
+             degrees_of_freedom * n_features * .5 * math.log(2.) +
+             np.sum(gammaln(.5 * (degrees_of_freedom -
+                                  np.arange(0, n_features)[:,
+                                                           np.newaxis])), 0))
+
+
+class BayesianGaussianMixture(BaseMixture):
+    """Variational estimation of a Gaussian mixture.
+
+    This class allows to infer an approximate posterior distribution over the
+    parameters of a Gaussian mixture distribution. The number of components can
+    be inferred from the data.
+
+    Read more in the :ref:`User Guide <bgmm>`.
+
+    Parameters
+    ----------
+    n_components: int, defaults to 1.
+        The number of mixture components.
+
+    covariance_type : {'full', 'tied', 'diag', 'spherical'}, defaults to 'full'.
+        String describing the type of covariance parameters to use.
+        Must be one of::
+        'full' (each component has its own general covariance matrix).
```

Comment:
Why the term "general"? I would rather say: each component is parameterized by its own covariance matrix.